lib/src/metta/environment.rs

```diff
@@ -0,0 +1,193 @@
+
+use std::path::{Path, PathBuf};
+use std::io::Write;
+use std::fs;
+use std::borrow::Borrow;
+
+use directories::ProjectDirs;
+
+use crate::common;
+
+/// Contains state and platform interfaces shared by all MeTTa runners.  This includes config settings
+/// and logger
+///
+/// Generally there will be only one environment object needed, and it can be accessed by calling the [platform_env] method
+#[derive(Debug)]
+pub struct Environment {
+    config_dir: Option<PathBuf>,
+    init_metta_path: Option<PathBuf>,
+    working_dir: Option<PathBuf>,
+    extra_include_paths: Vec<PathBuf>,
+}
+
+const DEFAULT_INIT_METTA: &[u8] = include_bytes!("init.default.metta");
+
+static PLATFORM_ENV: std::sync::OnceLock<Environment> = std::sync::OnceLock::new();
```

Comment:
Right.  The idea is to make it low-friction for the caller.  So the caller doesn't need to keep track of the environment as an object.

A caller can initialize the platform environment explicitly as the first thing they do.  Otherwise, the default environment will be automatically initialized lazily the first time it's needed.

As for unit tests, we could init the environment with the `no_cfg_dir` option, which would prevent the any `init.metta` file from running.  But doing that at the top of every test is a little tedious.  One option is to make a test preamble that we always run at the top of every test.  Another option is to make a test-specific `Metta::new_rust` function.

I think I'd prefer the test preamble approach because it lets us handle other places where the test environment differs from normal execution.

Reply:
> The idea is to make it low-friction for the caller. So the caller doesn't need to keep track of the environment as an object.
> A caller can initialize the platform environment explicitly as the first thing they do. Otherwise, the default environment will be automatically initialized lazily the first time it's needed.

The straightforward (not necessary right) way to do this is to make an instance of the `Environment` a part of the `MeTTa` instance. Which disadvantages do you see in this approach?

> But doing that at the top of every test is a little tedious. One option is to make a test preamble that we always run at the top of every test. Another option is to make a test-specific Metta::new_rust function.

Yes, I agree doing this in each test is too verbose. First thing which comes into my mind is to make a test-specific `Metta::new_rust()` function. I am not sure what is preamble approach. Could you please share a link to some example of this?

Reply:
It's a non-issue now, after the refactor in https://github.com/trueagi-io/hyperon-experimental/commit/dc15fc359ebb7b7dad9014891da563a6dfda709d the MeTTa runners can be initialized with a test environment.